website/source/guides/identity/approle-trusted-entities.html.md

```diff
@@ -0,0 +1,779 @@
+or deliver the `RoleID` and `SecretID` to our target systems.
+
+
+## Solution
+
+Use _Trusted Entities_ to deliver the AppRole authentication values. For
+example, use Terraform to deliver your `RoleID` or embed it into your AMI or
+Dockerfile. Then you might use Jenkins or Chef to obtain the
+[response-wrapped](/guides/secret-mgmt/cubbyhole.html) `SecretID` and deliver it
+to the target system.
+
+AppRole allows us to securely introduce the authentication token to the target
+system by preventing any single system from having full access to an
+authentication token that does not belong to. This helps us maintain the
+security principles of **least privilege** and **non-repudiation**.
+
+The important thing to note here is that regardless of what systems are
+considered as _Trusted Entities_, the same pattern applies.
+
+For example:
+
+- With Chef, you might use the [Vault Ruby Gem](https://github.com/hashicorp/vault-ruby)
+  for simplified interaction with Vault APIs
+- Terraform provides a Vault provider: [Provider: Vault - Terraform by HashiCorp](https://www.terraform.io/docs/providers/vault/index.html)
+- For Jenkins, you might use the Vault CLI or APIs directly, as described here:
+  [Reading Vault Secrets in your Jenkins pipeline](http://nicolas.corrarello.com/general/vault/security/ci/2017/04/23/Reading-Vault-Secrets-in-your-Jenkins-pipeline.html)
+
+
+## Prerequisites
+
+This guide assumes that you are proficient enough to perform basic Terraform
+tasks. If you are not familiar with Terraform, refer to the [online
+documentation](https://www.terraform.io/intro/getting-started/install.html).
+
+The following AWS resources are required to perform this demo:
+
+- An [Amazon S3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/gsg/CreatingABucket.html)
+- An [IAM user credential with administrator permissions](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_change-permissions.html)
+(to be able to create additional IAM policies and instance profiles)
+
+### Download demo assets
+
+Clone or download the demo assets from the
+[hashicorp/vault-guides](https://github.com/hashicorp/vault-guides/tree/master/identity/vault-chef-approle)
+GitHub repository to perform the steps described in this guide.
+
+The following assets can be found in the repository:
+
+- Chef cookbook (**`/chef`**): A sample cookbook with a recipe that installs NGINX
+and demonstrates Vault Ruby Gem functionality used to interact with Vault APIs.
+- Terraform configurations (**`/terraform-aws`**):
+    - **`/terraform-aws/mgmt-node`**: Configuration to set up a management
+    server running both Vault and Chef Server, for demo purposes.
+    - **`/terraform-aws/chef-node`**: Configuration to set up a Chef node and
+    bootstrap it with the Chef Server, passing in Vault's AppRole RoleID and the
+    appropriate Chef run-list.
+- Vault configuration (**`/vault`**): Data scripts used to configure the
```

Comment:
It doesn't look like the `/vault` folder exists

Reply:
Thank you for catching my typo.  This should've been `scripts`.